pydicom/uid.py

```diff
@@ -56,24 +63,40 @@ def __new__(cls, val):
 
         raise TypeError("UID must be a string")
 
-    def __str__(self):
-        """Return the human-friendly name for this UID"""
-        return self.name
-
     def __eq__(self, other):
         """Return True if `self` or `self.name` equals `other`."""
+        # TODO: v1.2 - The __ne__ override is deprecated
         if str.__eq__(self, other) is True:  # 'is True' needed (issue 96)
             return True
+
         if str.__eq__(self.name, other) is True:  # 'is True' needed (issue 96)
+            msg = "The equality test for \"UID == '{0}'\" is deprecated and " \
+                  "will be removed in pydicom v1.2. In the future use " \
+                  "\"UID.name == '{0}'\"".format(other)
+            warnings.warn(msg, DeprecationWarning)
             return True
+
```

Comment:
That would not warn in the case most likely overlooked after the change.
Maybe do instead:
```python 
if self and '.' not in self:
    warning.warn(...)
return __eq__(self.name, other) is True
```
This would be slightly slower, but I think we can live with that.

Reply:
Good suggestion